src/types/ComponentType.ts

```diff
@@ -85,17 +103,99 @@ export class ComponentType extends InheritableType {
+            for (const paramType of paramTypes) {
+                originalTypesToCheck.add(paramType);
+            }
+        }
+        if (funcType.returnType) {
+            originalTypesToCheck.add(funcType.returnType);
+        }
+        const additionalTypesToCheck = new Set<BscType>();
+        /*function addSubTypes(type: BscType) {
+            const subSymbols = type.getMemberTable().getAllSymbols(SymbolTypeFlag.runtime);
+            for (const subSymbol of subSymbols) {
+                if (!subSymbol.type.isBuiltIn && !(additionalTypesToCheck.has(subSymbol.type) || originalTypesToCheck.has(subSymbol.type))) {
+                    // if this is a custom type, and we haven't added it to the types to check to see if can add it to the additional types
+                    // add the type, and investigate any members
+                    additionalTypesToCheck.add(subSymbol.type);
+                    addSubTypes(subSymbol.type);
+                }
+
+            }
+        }*/
 
-    addCallFuncMember(name: string, data: ExtraSymbolData, type: BaseFunctionType, flags: SymbolTypeFlag) {
-        this.callFuncMemberTable.addSymbol(name, data, type, flags);
+        for (const type of originalTypesToCheck) {
+            if (!type.isBuiltIn) {
+                util.getCustomTypesInSymbolTree(additionalTypesToCheck, type, (subSymbol: BscSymbol) => {
+                    return !originalTypesToCheck.has(subSymbol.type);
+                });
+
+
+                //addSubTypes(type);
+            }
+        }
+
+        for (const type of [...originalTypesToCheck.values(), ...additionalTypesToCheck.values()]) {
+            if (!isPrimitiveType(type) && type.isResolvable()) {
+                // This type is a reference type, but was able to be resolved here
+                // add it to the table of associated types, so it can be used through a callfunc
+                const extraData = {};
+                if (associatedTypesTableProvider) {
+                    associatedTypesTableProvider().getSymbolType(type.toString(), { flags: SymbolTypeFlag.typetime, data: extraData });
+                }
+                let targetType = isAnyReferenceType(type) ? type.getTarget?.() : type;
+
+                this.callFuncAssociatedTypesTable.addSymbol(type.toString(), { ...extraData, isFromCallFunc: true }, targetType, SymbolTypeFlag.typetime);
+            }
+        }
+
+        // add this function to be available through callfunc
+        this.callFuncMemberTable.addSymbol(name, data, funcType, flags);
     }
 
     getCallFuncTable() {
         return this.callFuncMemberTable;
     }
 
     getCallFuncType(name: string, options: GetSymbolTypeOptions) {
-        return this.callFuncMemberTable.getSymbolType(name, options);
+        const callFuncType = this.callFuncMemberTable.getSymbolType(name, options);
+
+        const addAssociatedTypesTableAsSiblingToMemberTable = (type: BscType) => {
```

Comment:
if you find a callFunc for this a given name, adds a sibling table of the types required for that callFunc to the type.
This makes it so the required types for the callfunc are available in a different context